From: Jeff Mahoney <jeffm@suse.com>
Subject: overlayfs: compat, use correct dentry to detect compat mode in ovl_compat_is_whiteout
References: bsc#1032400
Patch-mainline: never, depends on overlayfs compat mode

We use dentry to test whether overlayfs is in compat mode but that
belongs to the upper mount.  We need to use ovl_dir to get the
overlayfs state.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 fs/overlayfs/compat.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/overlayfs/compat.c
+++ b/fs/overlayfs/compat.c
@@ -94,7 +94,7 @@ bool ovl_compat_is_whiteout(struct dentr
 	int res;
 	char val;
 
-	if (!ovl_compat_mode(dentry))
+	if (!ovl_compat_mode(ovl_dir))
 		return false;
 
 	if (!dentry)
